crates/iceberg/src/spec/manifest_list.rs

```diff
@@ -30,6 +30,9 @@ use self::{
 
 use super::{FormatVersion, StructType};
 
+/// The seq number when no added files are present.
+pub const UNASSIGNED_SEQ_NUMBER: i64 = -1;
```

Comment:
It's for [min_sequence_number](https://iceberg.apache.org/spec/#manifests:~:text=516-,min_sequence_number,-long). Which means that all manifest entry is deleted. Actually, I am not sure whether it's the correct process way because there is not `UNASSIGNED_SEQ_NUMBER` mentioned in spec.🤔 Does pyiceberg use `UNASSIGNED_SEQ_NUMBER`? cc @Fokko

Reply:
I see. Maybe we don't need this, and using `Option` is more appropriate?

Reply:
In PyIceberg we set it to `0` when it is not set (v2) or unknown (v1). It is used to effectively prune and delete files that are not relevant to the data that are being read.

In PyIceberg we first do the normal query planning by applying the partition filtering and the metrics. The new end up with a list of files where we compute the minimal data file sequence number:

https://github.com/apache/iceberg-python/blob/8c8abb5c4c258e32941110a9ce0938e1328290b3/pyiceberg/table/__init__.py#L1028-L1037

There is an obvious fallback to `INITIAL_SEQUENCE_NUMBER` which is `0`. If this happens then we know that we can't use this number to prune, and all the deletes files that are present will be included (because the sequence number there is also greater than or equal to zero.

I would suggest setting this to zero.

Reply:
LGTM.

Reply:
I guess for v2, here the min sequence number should be set as sequence number.
https://github.com/apache/iceberg-python/blob/8c8abb5c4c258e32941110a9ce0938e1328290b3/pyiceberg/manifest.py#L934C1-L934C1

> The minimum data sequence number of all live data or delete files in the manifest; use 0 when reading v1 manifest lists

I find that the minimum data sequence will be set if it the manifest has live files(the file exist before), otherwise it will be set as sequence number in manifest list writer.
https://github.com/apache/iceberg-python/blob/8c8abb5c4c258e32941110a9ce0938e1328290b3/pyiceberg/manifest.py#L733C1-L733C1

And I realize that I should not put the sequence number in manifest writer. The sequence number only need set in manifest list. For manifest entry, [the sequence number should be in read process](https://iceberg.apache.org/spec/#sequence-number-inheritance).

Reply:
I have add a commit to fix it. PTAL @liurenjie1024  @Fokko